### DISCUSSION TITLE
Why do we use zephyr_include_directories in Zephyr instead of zephyr_system_include_directories?

### DISCUSSION THREAD

**Discussion by silverv** [zephyrproject-rtos/zephyr#58810] [Q&A]
Is there a rationale?

**Comment by pavel-kirienko** [comment]
To expand upon the OPs question: this circumstance makes it difficult to perform static analysis on the application as static analyzers assume headers added via `-I` to be part of the analyzed codebase. The correct approach is to use `-isystem` which signals analysis tools that the referenced path is not part of the analyzed project and any compliance issues found there should be ignored.

The use of `zephyr_include_directories` instead of `zephyr_system_include_directories` for system headers appears to be a mistake unless we're missing something.
